NN_2L2H2O_poly2d/NN_2L2H2O_poly2d.cu

```diff
@@ -0,0 +1,146 @@
+     T* data = nullptr;     
+     
+     hsize_t bias_rank=0;
+     hsize_t* bias_dims = nullptr;
+     T* bias = nullptr;
+     
+     Layer_Net_t<T> layers;
+     
+     // Open HDF5 file handle, read only
+     H5File file(filename,H5F_ACC_RDONLY);
+     
+     
+     try{     
+          // Get saved layer names
+          vector<string> layernames;
+          layernames = Read_Attr_Data_By_Seq(file,PATHTOMODEL, LAYERNAMES); 
+
+          for (string layername : layernames) {
+               // for one single layer
+               // layer's fullpath
+               string layerpath = mkpath ( string(PATHTOMODEL),  layername ) ;
+               
+               // get this layer's dataset names
+               vector<string> weights;
+               weights = Read_Attr_Data_By_Seq(file,layerpath.c_str(), WEIGHTNAMES);
+               
+               
+               cout << " Reading out layer data: " << layername << endl;
+               for (string wt : weights ) {
+                    // foe one data set
+                    // dataset's path
+                    string datasetPath = mkpath(layerpath,wt) ;
+                    
+                    // check the dataset name's last character to see if this dataset is a Weight or a Bias
+                    if (wt.compare((wt.length()-1),1, checkchar )==0){
+                         // get out weight data
+                         Read_Layer_Data_By_DatName<T> (file, datasetPath.c_str(), data, data_rank, data_dims); 
+                    }else{
+                         // get out bias data
+                         Read_Layer_Data_By_DatName<T> (file, datasetPath.c_str(), bias, bias_rank, bias_dims);             
+                    }
+               }
+               // When reading out a dense layer, a 2d weight matrix is obtained
+               // Otherwise, it is a 0d matrix (null)
+               if (data_rank==2){
+                    cout << " Initialize dense layer : " << layername << endl;
+                    layers.insert_layer(layername, data_dims[0], data_dims[1], data, bias);
+                    data_rank=0;
+                    bias_rank=0;
+               } else {
+                    cout << " Initialize activiation layer : " << layername << endl;
+                    layers.insert_layer(layername, ACTTANH);               
+               }
+
+               
+               cout << " Layer " << layername << " is initialized. " <<endl <<endl;
+          }
+          
+          cout << "Inserting Layers finished !" <<endl;
+          
+          // In our test model, we insert 5 (fully_connect + tanh_activiation) layers
+          // plus 1 (fully_connect + linear_activiation) layers
+          // So change the last activiation layer's type to linear
+          layers.get_layer_by_seq(LASTATVID) -> acttype = ACTLINEAR;
+          
+          cout << endl;
+          cout << "Prediction all samples : " <<endl;
+          layers.predict(input, SAMPLECOUNT, SAMPLEDIM);
```

Comment:
very nice @yazhai , I like a lot how this file is short and simple to understand!!